_posts/2023-02-10-hippo-to-h3.md

```diff
@@ -0,0 +1,772 @@
+
+Taking this basis and measure family, the authors show that the discretized ODE for HiPPO-LegT takes the form of a state space model with the following transition matrices:
+
+$$A_{nk} = \frac{1}{\theta}\begin{cases} 
+(-1)^{n-k}(2n+1) & n \geq k \\
+2n+1 & n \leq k
+\end{cases}$$
+x
+$$B_n = \frac 1 \theta (2n+1)(-1)^n$$
+
+In *other* other words, this just corresponds to a simple sliding window average with equal weights. This naturally motivates the question: what about other types of moving averages? What happens, for example, if we use exponential weighting instead of uniform? But more on that later. First, let's point out some related work:
+
+#### The Legendre Memory Unit
+{% include figure.html path="assets/img/2023-02-10-hippo-to-h3/spiking.png" class="img-fluid" %}
+
+The HiPPO-LegT state space model parameterization is equivalent to that of the Legendre Memory Unit <d-cite key='lmu'></d-cite> (LMU) <d-cite key='lmu'></d-cite>, which was originally designed to mimic dynamics of biological Spiking Neural Networks. Gu et al. cite the LMU as motivation for HiPPO, but also note that the LMU “approaches the problem from the opposite direction as us; it considers approximating spiking neurons in the frequency domain, while we directly solve an interpretable optimization problem in the time domain"<d-cite key='hippo'></d-cite>. 
+
+### HiPPO-LagT: Laguerre Time-invariant
+{% include figure.html path="assets/img/2023-02-10-hippo-to-h3/lagt.png" class="img-fluid" %}
+
+As noted, the HiPPO-LegT model is determined by the choice of a uniform measure that acts like a sliding window average. We can construct the HiPPO operator that corresponds to exponential moving averages as well. In this case, the measure is given by the following:
+
+$$
+\mu^{(t)}(x) = e ^ {x - t} \mathbb I_{\lbrack -\infty, t\rbrack}(x). 
+$$
+
+To find the corresponding state space model, we need to first find a polynomial basis that is orthonormal with respect to this measure. Luckily, this is precisely what the **Laguerre polynomials** are. The Laguerre polynomials are defined as the system of polynomials that satisfy the following:
+
+1. They are orthogonal on the interval $$[0, \infty)$$ with respect to the measure $$\mu^{(t)}(x) = e ^ {x - t} \mathbb I_{\lbrack -\infty, t\rbrack}(x)$$.
+1. Denote by $$L_n$$ the nth Laguerre polynomial. Then, for all $$n$$, $$L_n(0) = 1$$ and $$L_n'(0) = 0$$.
+
+The transition matrices for the HiPPO-LagT state space model are given by:
+
+$$
+A_{nk} = \begin{cases} 
+1 & n \geq k \\
+0 & n \leq k
+\end{cases}
+$$
+
+$$
+B_n = 1
+$$
+
+### HiPPO-LegS: Legendre Scale-invariant
+{% include figure.html path="assets/img/2023-02-10-hippo-to-h3/legs.png" class="img-fluid" %}
```

Comment:
I think this needs to be `... path="assets/img/2023-02-10-hippo-to-h3/legs.png ...`

Reply:
I'm not sure if I follow... those look like the same path to me. 

But I noticed the .png in assets capitalized the s in "legS" in the filename. It didn't seem to cause any issues on the post, as far as case-sensitivity, but I updated it to match.

Reply:
Sorry that's what I meant/thought I wrote. If you're using a Mac then I think capitalization doesn't matter, but it does on linux. In any case thanks I think this should be good!